src/test/java/tatracker/model/group/GroupTest.java

```diff
@@ -0,0 +1,178 @@
+        assertFalse(NO_STUDENTS.equals(MANY_STUDENTS));
+    }
+
+    @Test
+    public void setIdentifier() {
+        Group group = new Group("G03");
+        group.setIdentifier("G04");
+
+        //Identifier value should change
+        assertTrue(group.getIdentifier().equals("G04"));
+    }
+
+    @Test
+    public void setGroupType() {
+        Group group = new Group("G03", GroupType.LAB);
+        group.setGroupType(GroupType.TUTORIAL);
+
+        //GroupType should change
+        assertTrue(group.getGroupType().equals(GroupType.TUTORIAL));
+    }
+
+    @Test
+    public void sortStudentsAlphabetically() {
+        Group group = MANY_STUDENTS;
+        group.sortStudentsAlphabetically();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.ALICE);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.DANIEL);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.GEORGE);
+    }
+
+    @Test
+    public void sortStudentsByRatingAscending() {
+        Group group = MANY_STUDENTS;
+        group.sortStudentsByRatingAscending();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.ALICE);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.BENSON);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.ELLE);
+    }
+
+    @Test
+    public void sortStudentsByRatingDescending() {
+        Group group = MANY_STUDENTS;
+        group.sortStudentsByRatingDescending();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.ELLE);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.BENSON);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.FIONA);
+    }
+
+    @Test
+    public void sortStudentsByMatricNumber() {
+        Group group = MANY_STUDENTS;
+        group.sortStudentsByMatricNumber();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.BENSON);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.ELLE);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.FIONA);
+    }
+
+    @Test
+    public void hasStudent() {
+        //First student in student list
+        assertTrue(MANY_STUDENTS.hasStudent(TypicalStudents.ALICE));
+
+        //Middle of the list
+        assertTrue(MANY_STUDENTS.hasStudent(TypicalStudents.ELLE));
+
+        //End of the list
+        assertTrue(MANY_STUDENTS.hasStudent(TypicalStudents.DANIEL));
+
+        //Not in the list
+        assertFalse(MANY_STUDENTS.hasStudent(TypicalStudents.BOB));
+
+        //List has no students
+        assertFalse(NO_STUDENTS.hasStudent(TypicalStudents.ALICE));
+
+        //Only one in the list
+        assertTrue(ONE_STUDENT.hasStudent(TypicalStudents.ALICE));
+
+        //Not in the list
+        assertFalse(ONE_STUDENT.hasStudent(TypicalStudents.BOB));
```

Comment:
Not right boundaries. It should be:
* has student
- one student
- many students

* does not have student
 - empty list
- one student
- many students
- **null**

So 1 and 2 are the same

Reply:
Need null check